builder/builder-next/exporter/writer.go

```diff
@@ -173,33 +175,39 @@ func normalizeLayersAndHistory(diffs []digest.Digest, history []ocispec.History,
 
 type refMetadata struct {
 	description string
-	createdAt   time.Time
+	createdAt   *time.Time
 }
 
 func getRefMetadata(ref cache.ImmutableRef, limit int) []refMetadata {
-	if limit <= 0 {
-		return nil
-	}
-	meta := refMetadata{
-		description: "created by buildkit", // shouldn't be shown but don't fail build
-		createdAt:   time.Now(),
-	}
 	if ref == nil {
-		return append(getRefMetadata(nil, limit-1), meta)
+		return make([]refMetadata, limit)
 	}
-	if descr := cache.GetDescription(ref.Metadata()); descr != "" {
-		meta.description = descr
+
+	layerChain := ref.LayerChain()
+	defer layerChain.Release(context.TODO())
+
+	if limit < len(layerChain) {
+		layerChain = layerChain[len(layerChain)-limit:]
 	}
-	meta.createdAt = cache.GetCreatedAt(ref.Metadata())
-	p := ref.Parent()
-	if p != nil {
-		defer p.Release(context.TODO())
+
+	metas := make([]refMetadata, len(layerChain))
+	for i, layer := range layerChain {
+		meta := &metas[i]
+
+		if description := layer.GetDescription(); description != "" {
+			meta.description = description
+		} else {
+			meta.description = "created by buildkit" // shouldn't be shown but don't fail build
```

Comment:
Was wondering if we use this information anywhere (and if so, if this at some point should include the buildkit version)